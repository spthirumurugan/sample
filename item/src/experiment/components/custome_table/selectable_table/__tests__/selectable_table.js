import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import SelectableTable from '../selectable_table';
import dummyState from '../../__mock__/dummy_state';

const middlewares = [];
const mockStore = configureStore(middlewares);

jest.mock('../table_row', () => 'tr');
jest.mock('../table_header', () => 'th');

describe('components', () => {
  describe('<SelectableTable />', () => {
    it('should render', () => {
      const store = mockStore(dummyState);
      const comp = renderer.create(
        <SelectableTable store={store} viewId={'view_1'} datasetIds={['id1', 'id2']} />
      ).toJSON();
      expect(comp).toMatchSnapshot();
    });
  });
});
