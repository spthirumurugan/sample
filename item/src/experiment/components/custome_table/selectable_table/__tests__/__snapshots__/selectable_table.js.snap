// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <SelectableTable /> should render 1`] = `
<div>
  <table
    className="table table-bordered"
    onClick={[Function]}
  >
    <th
      viewId="view_1"
    />
    <tbody>
      <tr
        datasetId="id1"
        viewId="view_1"
      />
      <tr
        datasetId="id2"
        viewId="view_1"
      />
    </tbody>
  </table>
</div>
`;
